File: core/src/net/natruid/jungle/components/IndicatorComponent.java
package net.natruid.jungle.components;

import com.artemis.Component;
import com.artemis.annotations.EntityId;
import com.artemis.annotations.PooledWeaver;
import net.natruid.jungle.utils.IndicatorType;

@PooledWeaver
public class IndicatorComponent extends Component {
    @EntityId
    public int entityId = -1;
    public IndicatorType type = IndicatorType.MOVE_AREA;

    protected void reset() {
        type = IndicatorType.MOVE_AREA;
    }
}

File: core/src/net/natruid/jungle/components/IndicatorComponent.kt
package net.natruid.jungle.components

import com.artemis.Component
import com.artemis.annotations.EntityId
import net.natruid.jungle.utils.IndicatorType

class IndicatorComponent(
    @EntityId var entityId: Int = -1,
    var type: IndicatorType = IndicatorType.MOVE_AREA
) : Component()
